nomad/structs/structs.go

```diff
@@ -4720,6 +4734,7 @@ type JobListStub struct {
 	ModifyIndex       uint64
 	JobModifyIndex    uint64
 	SubmitTime        int64
+	Meta              map[string]string
```

Comment:
We should make sure this field is included on `api.JobListStub` as well. 

We should probably add a `api.JobListWithFields` method to the `api` package as well (under api/jobs.go), but I'm fine with shipping this PR without and we can come back to that later: right now the UI will be the only consumer of this field.